data-prepper-plugins/otel-trace-group-prepper/src/test/java/com/amazon/dataprepper/plugins/prepper/oteltracegroup/OTelTraceGroupPrepperTests.java

```diff
@@ -127,6 +134,9 @@ public void testTraceGroupFillSuccess() throws IOException {
         // Arrange
         Record<String> testRecord = buildRawSpanRecord(TEST_RAW_SPAN_MISSING_TRACE_GROUP_JSON_FILE_1);
         List<Record<String>> testRecords = Collections.singletonList(testRecord);
+        final List<Measurement> spansMissingTraceGroupMeasures = MetricsTestUtil.getMeasurementList(
+                new StringJoiner(MetricNames.DELIMITER).add(TEST_PIPELINE_NAME).add(PLUGIN_NAME)
+                        .add(OTelTraceGroupPrepper.SPANS_MISSING_TRACE_GROUP_INFO).toString());
```

Comment:
nit - I see 40+ usages of `getMeasurementList()` all requiring users to create a StringJoiner and know how to assemble the string. Might be time to create utility method which takes in pipeline/plugin/metric Strings, does the joining, and returns the measurement object(s).

This highlighted block could also be split out into a private method as it's repeated 4 times in this class.

Reply:
I agree. Let me work on the second part (highlighted block splitting) and reserve the first in a separate PR for cleaner workflow path.